### ISSUE TITLE
intermittent test_mail process failure[PATCH]

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7591]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#342|
|Type|defect|
|Created|2003-10-16 00:43:51Z|

<details><summary>Searchable metadata</summary>

```
trac-id__342 342
type__defect defect
reporter__warner warner
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066265031000000 1066265031000000
changetime__1468715099897402 1468715099897402
version__ 
owner__ 
cc__spiv cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
I've seen this happen twice in the full-2.3 builder.
I suspect it is related to #337 (test_process error).

ERROR: testProcessAlias (twisted.test.test_mail.ProcessAliasTestCase)
-------------------------------------------------------------------------------
"/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/test/test_mail.py", line
1029, in testProcessAlias
    unittest.deferredResult(m.eomReceived())
  File "/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/trial/util.py",
line 72, in deferredResult
    raise result
Failure: [Failure instance: Traceback: twisted.internet.defer.TimeoutError,
Callback timed out
]
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Here's another failure for the same test, from the freebsd slave:

===============================================================================
FAILURE: testCyclicAlias (twisted.test.test_mail.ProcessAliasTestCase)
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
246, in _runPhase
    stage(*args, **kwargs)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/util.py", line 43,
in reactorCleanUp
    raise unittest.FailTest, msg
FailTest: 
pendingTimedCalls still pending:
 &lt;DelayedCall 145373580 [-0.00216996669769s] called=0 cancelled=0 &lt;code object
reapAllProcesses at 0x83b06a0, file
"/home/slave/twisted/bsd-full2.2/Twisted/twisted/internet/process.py", line 64>()>

-------------------------------------------------------------------------------
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
My guess is that the reactor doesn't have time to clean up after itself.
Here's a patch which should fix the problem:

Index: twisted/test/test_mail.py
===================================================================
RCS file: /cvs/Twisted/twisted/test/test_mail.py,v
retrieving revision 1.33
diff -u -r1.33 test_mail.py
--- twisted/test/test_mail.py   9 Oct 2003 17:32:28 -0000       1.33
+++ twisted/test/test_mail.py   21 Oct 2003 06:29:43 -0000
@@ -1099,6 +1099,7 @@
             mail.alias.MessageWrapper(p, 'echo')
         ])
         self.assertEquals(r, expected)
+        reactor.iterate(1)
 if not components.implements(reactor, interfaces.IReactorProcess):
     ProcessAliasTestCase = "IReactorProcess not supported"
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
I tried the solution, lets see if it helps.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
It seems this problem (and 337) were fixed.
Let's give it a couple more tests, and close it.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
This bug, and 337, seem gone.
Please open new issues for intermittent failures if they show
up again.
Thanks
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub commented|
|-|-|

In changeset dae42dcba15e0ecc5b16f3d104b52cf7ca2f4051
```
#!CommitTicketReference repository="" revision="dae42dcba15e0ecc5b16f3d104b52cf7ca2f4051"
Merge pull request #342 from twisted/8577-rodrigc-win-trial-2

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8577

Remove "python" before invoking tests via "tox -e py35-tests"
```
